farm/infer.py

```diff
@@ -251,6 +251,13 @@ def inference_from_dicts(self, dicts, rest_api_schema=False, max_processes=128):
                 f"b) ... run inference on a downstream task: make sure your model path {self.name} contains a saved prediction head"
             )
 
+        # Assign a sequential doc_id to each dictionary
```

Comment:
Shouldn't this only be executed for rest_api_schema=true? For SQuAD input format we want to use the passed doc id.

Reply:
Actually there's a better way to do this now which is that the doc id for squad or rest api format is inferred from basket.id